addons/website/models/website_menu.py

```diff
@@ -26,6 +26,7 @@ def _default_sequence(self):
     child_id = fields.One2many('website.menu', 'parent_id', string='Child Menus')
     parent_path = fields.Char(index=True)
     is_visible = fields.Boolean(compute='_compute_visible', string='Is Visible')
+    group_ids = fields.Many2many('res.groups', index=True, string='Groups')
```

Comment:
@Yenthe666  Do we really need `index=True` on this `Many2many` field ?

Reply:
@sswapnesh I recently went on a trip to Odoo SA and @nseinlet told me the use of indexes. As I assume this will be used quite a lot and will probably hold quite some records I think it is a good idea. Perhaps @nseinlet can give his opinion here.

Reply:
Indeed, Its good to use Index on such fields but I am not sure If it has any effect on many2many field or not.

Reply:
Index have no effect on : company_depend, One2many, Many2Many, non-stored fields.
But the relational table of many2many is an indexed table.

For On2many field it is important to put index on the inverse_name field.
Exemple :
```python
PurchaseOrder():
      line_ids = fields.One2many('purchase.order.line','order_id')

PurchaseOrderLine():
     order_id = fields.Many2one('purchase.order', index=True)
```

Note : One2many field is such a computed field with this function :
`return self.env['purchase.order.line'].search([('id','=',self.order_id.id)])`

How to use index :
If you use lot of time `self.env['purchase.order.line'].search([('product_id','in', blabla.ids)])`, it is recommended to put index=True for product_id field, this increase a lot the performance.
On a field never used in search methode, it is not usefull to put index. (put index increase the size of database and memory used).

Reply:
:+1: 

> But the relational table of many2many is an indexed table.

It adds Index at  https://github.com/odoo/odoo/blob/12.0/odoo/fields.py#L2630

We can also add Index on other (Char,Int) fields, if they are used at multiple places in domain.

https://github.com/odoo/odoo/pull/24252#issuecomment-382340649 is also  Good read on same.

Reply:
cc : https://github.com/odoo/odoo/pull/32606

Reply:
Hmm that where indeed some interesting reads. Maybe it is better not to set one after all.